etl/steps/data/meadow/wb/2022/wb_gender.py

```diff
@@ -0,0 +1,258 @@
+        msk = merged["variable"] != merged["indicator_name"]
+        x = merged.loc[msk, ["variable", "indicator_name"]].sort_values("variable")
+        assert x.shape == (45, 2,), (
+            "There are 45 expected variables to miss-match namings between data and metadata file, but a different"
+            f" amount was found {x.shape}!"
+        )
+
+    def final_formatting(self, df: pd.DataFrame, metadata: pd.DataFrame) -> pd.DataFrame:
+        """Final formatting of the data.
+
+        To finalise the formatting of the data, we replace the variable names with those from metadata.
+        In addition, we sort the rows and columns.
+
+        Parameters
+        ----------
+        df : pd.DataFrame
+            _description_
+        metadata : pd.DataFrame
+            _description_
+
+        Returns
+        -------
+        pd.DataFrame
+            _description_
+        """
+        # Use variable names from metadata
+        log.info("Replacing variable names with those from metadata...")
+        df = df.merge(
+            metadata[["series_code", "indicator_name"]], left_on="variable_code", right_on="series_code"
+        ).reset_index(drop=True)
+        df = df[["location", "indicator_name", "year", "value"]].rename(columns={"indicator_name": "variable"})
+        # Sort dataframe and set index
+        log.info("Sorting dataframe and setting index...")
+        df = df.sort_values(self.columns_idx).set_index(self.columns_idx)
+        return df
+
+    def run(self, dest_dir: str) -> None:
+        """Run Meadow step."""
+        # Load data and metadata
+        df, metadata = self.load_data_from_walden()
+        # Format metadata
+        metadata = metadata.pipe(self.format_metadata)
+        # Format data
+        df = df.pipe(self.format_data, metadata)
+        # Initiate dataset
+        ds = init_dataset(dest_dir, self.walden_ds)
+        # Add tables to dataset
+        ds = add_tables_to_ds(ds, df, metadata)
+
+
+def _clean_variable_name(text: str) -> str:
+    """Clean string."""
+    return re.sub(r"\s+", " ", text).replace(" 00", " 0")
+
+
+def init_dataset(dest_dir: str, walden_ds: walden.Dataset) -> Dataset:
+    "Initialise Meadow dataset."
+    log.info("Initialising Meadow dataset...")
+    ds = Dataset.create_empty(dest_dir)
+    ds.metadata = convert_walden_metadata(walden_ds)
+    ds.metadata.short_name = "wb_gender"
+    ds.save()
+    return ds
+
+
+def add_tables_to_ds(ds: Dataset, df: pd.DataFrame, metadata: pd.DataFrame) -> Dataset:
+    """Add tables to Meadow dataset"""
+    log.info("Adding tables to Meadow dataset...")
+    tables = [
+        (df, "data"),
+        (metadata, "metadata"),
+    ]
+    for t in tables:
+        table = Table(t[0])
+        table.metadata.short_name = t[1]
+        ds.add(utils.underscore_table(table))
+    return ds
+
+
+def run(dest_dir: str) -> None:
+    """Run pipeline."""
+    WBGenderMeadowStep().run(dest_dir)
```

Comment:
I'd personally prefer to have the `run` function in a data step as explicit as possible. So, if anyone wants to quickly have an overview of what's done to the data, they should be able to check the content of `run` and get a sense of it. So, at the very least, I'd recommend moving the content of the `run` method of `WBGenderMeadowStep` into the global `run` function. This way you can still see:
```
# Load data and metadata
df, metadata = self.load_data_from_walden()
...
```
Feel free to ignore my suggestion if you disagree.

Reply:
makes sense, thanks!